pkg/yamlSchemaValidator/validator_test.go

```diff
@@ -0,0 +1,112 @@
+package yamlSchemaValidator
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/datreeio/datree/pkg/fileReader"
+	"github.com/datreeio/datree/pkg/policy"
+	"gopkg.in/yaml.v2"
+)
+
+type TestFilesByRuleId = map[int]*FailAndPassTests
+
+type FailAndPassTests struct {
+	fails  []*FileWithName
+	passes []*FileWithName
+}
+
+type FileWithName struct {
+	name    string
+	content string
+}
+
+func TestDefaultRulesValidation(t *testing.T) {
+	err := os.Chdir("../../")
+	if err != nil {
+		panic(err)
+	}
+
+	defaultRules, err := policy.GetDefaultRules()
+	if err != nil {
+		panic(err)
+	}
+
+	testFilesByRuleId := getTestFilesByRuleId(t)
+	validator := New()
+
+	for _, rule := range defaultRules.Rules {
+		validatePassing(t, validator, rule.Schema, rule.ID, testFilesByRuleId[rule.ID].passes, true)
+		validatePassing(t, validator, rule.Schema, rule.ID, testFilesByRuleId[rule.ID].fails, false)
+	}
+}
+
+func validatePassing(t *testing.T, validator *YamlSchemaValidator, schemaContent map[string]interface{}, ruleId int, files []*FileWithName, expectPass bool) {
+	for _, file := range files {
+		schemaBytes, err := yaml.Marshal(schemaContent)
+		if err != nil {
+			panic(err)
+		}
+
+		res, err := validator.Validate(string(schemaBytes), file.content)
+		if err != nil {
+			panic(err)
+		}
+
+		if len(res.Errors()) > 0 && expectPass {
+			t.Errorf("Expected validation for rule with id %d to pass, but it failed for file %s\n", ruleId, file.name)
+		}
+		if len(res.Errors()) == 0 && !expectPass {
+			t.Errorf("Expected validation for rule with id %d to fail, but it passed for file %s\n", ruleId, file.name)
```

Comment:
This error is thrown from `gojsonschema`? If so, this is a bit coupling and might be unexpected in the future. I suggest creating a specific error and returning it. This way we can actually use the same error and it will make more sense why we expected this error.

Reply:
Generally I agree but in this case `Errors()` is a very generic method name and we can easily implement the same logic even if we decide to change the package we use.
I see no reason adding our own errors implementation when the existing one is readable.
Also, the test is not using the data inside the `Errors()` array but only checking its length